query/src/sql/exec/mod.rs

```diff
@@ -228,4 +239,80 @@ impl PipelineBuilder {
 
         Ok(output_schema)
     }
+
+    fn build_aggregate(
+        &mut self,
+        aggregate: &AggregatePlan,
+        input_schema: DataSchemaRef,
+    ) -> Result<DataSchemaRef> {
+        let mut agg_expressions = Vec::with_capacity(aggregate.agg_expr.len());
+        let expr_builder = ExpressionBuilder::create(&self.metadata);
+        for scalar_expr in aggregate.agg_expr.iter() {
+            let scalar = scalar_expr.as_any().downcast_ref::<Scalar>().unwrap();
+            let expr = expr_builder.build(scalar)?;
+            agg_expressions.push(expr);
+        }
+        let partial_schema = Arc::new(DataSchema::new(RewriteHelper::exprs_to_fields(
+            agg_expressions.as_slice(),
+            &input_schema,
+        )?));
+
+        dbg!(partial_schema.clone());
+
+        let mut group_expressions = Vec::with_capacity(aggregate.group_expr.len());
+        for scalar_expr in aggregate.group_expr.iter() {
+            let scalar = scalar_expr.as_any().downcast_ref::<Scalar>().unwrap();
+            let expr = expr_builder.build(scalar)?;
+            group_expressions.push(expr);
+        }
+        let mut final_exprs = agg_expressions.to_owned();
+        final_exprs.extend_from_slice(group_expressions.as_slice());
+        let final_schema = Arc::new(DataSchema::new(RewriteHelper::exprs_to_fields(
+            final_exprs.as_slice(),
+            &input_schema,
+        )?));
+
+        dbg!(final_schema.clone());
```

Comment:
I think this is only for debug, please tracing::debug! or removed the dbg probe.

Reply:
Yes, will remove